org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetManagerPage.java

```diff
@@ -58,29 +86,191 @@ public PresetManagerPage(PresetRepository repository) {
 
 	@Override
 	public void createControl(Composite parent) {
-		initializeDialogUnits(parent);
-
 		setTitle(MESSAGE_PRESET_MANAGER_PAGE_TITLE);
 		setDescription(MESSAGE_PRESET_MANAGER_PAGE_DESCRIPTION);
 
 		ScrolledComposite sc = new ScrolledComposite(parent, SWT.H_SCROLL | SWT.V_SCROLL);
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
+		container.setLayout(new GridLayout(2, false));
 
-		// TODO: create manager control here
-		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create manager control here");
+		createPresetTable(container);
+		createButtons(container);
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
+
+		bindListeners();
+	}
+
+	private void createPresetTable(Composite parent) {
+		tableViewer = new TableViewer(parent, SWT.V_SCROLL | SWT.BORDER | SWT.MULTI);
+		tableViewer.getControl().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		tableViewer.setContentProvider(new PresetTableContentProvider());
+		tableViewer.setLabelProvider(new DelegatingStyledCellLabelProvider(new PresetTableLabelProvider()));
+		tableViewer.setInput(repository);
+	}
+
+	private void createButtons(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		container.setLayoutData(new GridData(SWT.FILL, SWT.BEGINNING, false, true));
+		GridLayout layout = new GridLayout(1, true);
+		layout.marginHeight = 0;
+		layout.marginWidth = 0;
+		container.setLayout(layout);
+
+		newButton = createButton(container, MESSAGE_NEW_BUTTON);
+		editButton = createButton(container, MESSAGE_EDIT_BUTTON);
+		duplicateButton = createButton(container, MESSAGE_DUPLICATE_BUTTON);
+		removeButton = createButton(container, MESSAGE_REMOVE_BUTTON);
+		importButton = createButton(container, MESSAGE_IMPORT_FILES_BUTTON);
+		exportButton = createButton(container, MESSAGE_EXPORT_FILE_BUTTON);
+	}
+
+	private Button createButton(Composite parent, String text) {
+		Button button = new Button(parent, SWT.NONE);
+		button.setText(text);
+		button.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+		return button;
 	}
 
-	// TODO: call this function when "New" or "Edit" button clicked
 	private void openPresetEditingWizardFor(IPreset preset) {
-		if (DialogToolkit.openWizardWithHelp(new PresetEditingWizard(preset))) {
-			// TODO: save the modified preset to the repository
+		while (DialogToolkit.openWizardWithHelp(new PresetEditingWizard(preset))) {
+			try {
+				// TODO: properly save to repository
```

Comment:
As said in TODO, presets are not properly saved after they are created.  All of the fields are not repopulated correctly.

Reply:
Also, when you edit a file, it adds an additional file to the preset manager instead of simply editing the original file.